pkg/cache/v3/linear.go

```diff
@@ -21,46 +21,73 @@ import (
 // single collection indexed by resource names and manages resource versions
 // internally. It implements the cache interface for a single type URL and
 // should be combined with other caches via type URL muxing. It can be used to
 // supply EDS entries, for example, uniformly across a fleet of proxies.
 type LinearCache struct {
-	// Type URL specific to the cache.
+	// typeURL provides the type of resources managed by the cache.
+	// This information is used to reject requests watching another type, as well as to make
+	// decisions based on resource type (e.g. whether sotw must return full-state).
 	typeURL string
-	// Collection of resources indexed by name.
-	resources map[string]types.Resource
-	// Watches open by clients, indexed by resource name. Whenever resources
-	// are changed, the watch is triggered.
-	watches map[string]map[int64]ResponseWatch
-	// Set of watches for all resources in the collection, indexed by watch id.
-	// watch id is unique for sotw watches and is used to index them without requiring
-	// the watch itself to be hashable, as well as making logs easier to correlate.
-	watchAll map[int64]ResponseWatch
-	// Continuously incremented counter used to index sotw watches.
-	sotwWatchCount int64
-	// Set of delta watches. A delta watch always contain the list of subscribed resources
-	// together with its current version
-	// version and versionPrefix fields are ignored for delta watches, because we always generate the resource version.
-	deltaWatches map[int64]DeltaResponseWatch
-	// Continuously incremented counter used to index delta watches.
-	deltaWatchCount int64
-	// versionMap holds the current hash map of all resources in the cache when delta watches are present.
-	// versionMap is only to be used with delta xDS.
-	versionMap map[string]string
-	// Continuously incremented version.
+
+	// resources contains all resources currently set in the cache and associated versions.
+	resources map[string]cachedResource
+
+	// resourceWatches keeps track of watches currently opened specifically tracking a resource.
+	// It does not contain wildcard watches.
+	// It can contain resources not present in resources.
+	resourceWatches map[string]watches
+	// wildcardWatches keeps track of all wildcard watches currently opened.
+	wildcardWatches watches
+	// currentWatchID is used to index new watches.
+	currentWatchID uint64
+
+	// version is the current version of the cache. It is incremented each time resources are updated.
 	version uint64
-	// Version prefix to be sent to the clients
+	// versionPrefix is used to modify the version returned to clients, and can be used to uniquely identify
+	// cache instances and avoid issues of version reuse.
 	versionPrefix string
-	// Versions for each resource by name.
-	versionVector map[string]string
+
+	// computeResourceVersion indicates whether the cache is currently computing and storing stable resource versions.
+	computeResourceVersion bool
```

Comment:
It seems like in some cases this variable is used to indicate that you should be computing resource versions, e.g. in NewLinearCache, and in some cases it is used to indicate that no stable versions have been computed and you should compute them, e.g. in `CreateDeltaWatch`. 

Could we rename is variable to reference the intent of the system instead of an action?